### ISSUE TITLE
Add functions for reading annotated sleep data

### ISSUE THREAD

**Issue by hofaflo** [cbrnr/sleepecg#35]
The goal is to have functions which download and read all common polysomnography datasets. Each record will be stored in a dataclass containing heartbeat times and annotated sleep stages.

Each reader should be added via a separate PR, this issue gives an overview of the progress.

Datasets:
- [x] [MESA](https://sleepdata.org/datasets/mesa) (Multi-Ethnic Study of Atherosclerosis) #28 
- [x] [SHHS](https://sleepdata.org/datasets/shhs) (Sleep Heart Health Study) #48 
- [x] [SLPDB](https://physionet.org/content/slpdb/1.0.0/) (MIT-BIH Polysomnographic Database) #47
- [ ] [MASS](http://ceams-carsm.ca/en/mass/) (requires registration including project description and ethics approval, so this is not really open)
- [ ] [CAPSLPDB](https://physionet.org/content/capslpdb/1.0.0/) (CAP Sleep Database)
- [ ] [CinC 2018](https://physionet.org/content/challenge-2018/1.0.0/) (You Snooze You Win - The PhysioNet Computing in Cardiology Challenge 2018)
- [ ] [sleep-accel](https://physionet.org/content/sleep-accel/1.0.0/) (Motion and heart rate from a wrist-worn wearable and labeled sleep from polysomnography)
- [ ] [DREAMS](https://zenodo.org/record/2650142)
- [ ] [ISRUC-SLEEP Dataset](https://sleeptight.isr.uc.pt/)

Planned future extensions:
- metadata
- actigraphy data

**Comment by viranovskaya** [comment]
Hi, I’d like to work on the CAPSLPDB reader.

Before starting, I checked all 108 annotation tables and EDF headers, as well as the reference `ScoringReader.m`. There are a few dataset details worth agreeing on first:

- stages should be mapped from `Event`, not `Sleep Stage` (`n12` has a malformed stage cell but a valid `SLEEP-S3` event);
- 80 files contain missing 30-second rows, representing 689 epochs in total; `ScoringReader.m` fills these gaps with the previous stage;
- timestamps may use `:` or `.`, `Position` may be absent, `n16` uses `Duration [s]`, and `nfle27` ends with a truncated row;
- two annotation tables begin before the EDF and 20 extend slightly past it;
- `n16` has no ECG channel, while `nfle25` and `nfle33` contain separate `ECG1` and `ECG2` channels.

I propose that `read_capslpdb()` use the `.txt` annotation tables, map S3/S4 to N3 and movement to UNDEFINED, keep only complete epochs covered by the EDF, and fill internal gaps as `ScoringReader.m` does. Since the stage grid is not always aligned with EDF start, the returned `SleepRecord` would be cropped to the first retained scored epoch and heartbeat times rebased to that point. For `nfle25` and `nfle33`, I would use `ECG1 - ECG2`.

For `n16`, would you prefer yielding a stage-only record with `heartbeat_times=None` and a warning, or skipping it? If this direction looks right, I can add parser fixtures for each edge case and validate the reader end-to-end on `n12` before opening a PR.

**Comment by cbrnr** [comment]
Great, thank you! I agree with your suggestions. Regarding `n16`, I would tend to skip that record. I think warning if that record is requested would also be a good idea, what do you think?

**Comment by viranovskaya** [comment]
Yes, I agree. Skipping `n16` is cleaner because this reader is intended to return ECG-based records. I’ll add a warning whenever `n16` matches the requested pattern, so it is not skipped silently.

I’ll proceed with the parser fixtures and the end-to-end check on `n12`.
